src/tools/genpolicy/rules.rego

```diff
@@ -911,15 +911,13 @@ allow_storage_options(p_storage, i_storage, layer_ids, root_hashes) {
     print("allow_storage_options 4: start")
 
     p_storage.driver == "smb"
-    count(i_storage.options) == 8
-    i_storage.options[0] == "dir_mode=0666"
-    i_storage.options[1] == "file_mode=0666"
-    i_storage.options[2] == "mfsymlinks"    
-    i_storage.options[3] == "cache=strict"  
-    i_storage.options[4] == "nosharesock"
-    i_storage.options[5] == "actimeo=30"    
-    startswith(i_storage.options[6], "addr=")
-    creds = split(i_storage.options[7], ",")
+    count(i_storage.options) == count(p_storage.options) + 2
+
+    p_storage_matches := [i | i := idx; idx < count(p_storage.options); p_storage.options[idx] == i_storage.options[idx]]
```

Comment:
Please log a task as a reminder to add a test that shows this expression works properly - e.g., when Sumedh will make adding these tests easier.